### DISCUSSION TITLE
Question about sionna/ofdm/eqalization.py

### DISCUSSION THREAD

**Discussion by PudongDacihua** [NVlabs/sionna#206] [Q&A]
Hi, I'm using the oldest version of sionna to do some research about DeepRx, recently, I updated sionna to the newest version. I encountered some issues while using the new version, and I think it might be due to my incomplete understanding of the new version.
Here is the question:
 When running examples/Neural_Receiver.ipynb, an error 
"Call arguments received by layer 'lmmse_equalizer_6' (type LMMSEEqualizer):
      • inputs=['tf.Tensor(shape=(None, 1, 2, 14, 128), dtype=complex64)', 'tf.Tensor(shape=(None, 1, 2, 1, 1, 14, 116), dtype=complex64)', 'tf.Tensor(shape=(), dtype=float32)', 'tf.Tensor(shape=(None,), dtype=float32)']"
happened in:
# Dictionnary storing the evaluation results
BLER = {}
model = E2ESystem('baseline-perfect-csi')
_,bler = sim_ber(model, ebno_dbs, batch_size=128, num_target_block_errors=100, max_mc_iter=100)
BLER['baseline-perfect-csi'] = bler.numpy()

From the current perspective, the issue seems to be related to channel equalization. In the old version, only the LMMSEEqualizer was present. However, the new version introduces new equalization methods and modifies the code structure. The files provided in the new version can run in the old version, but not in the new version. The key difference is that many operations of LMMSEEqualizer in the new version have been integrated into OFDMEqualizer.

Does this mean that the new version requires calling OFDMEqualizer to process LMMSEEqualizer? However, this is not done in the example code. Could you please clarify this? Or it could be that there are still issues with my code version.
                                                                                                                                               Thanks!

**Comment by faycalaa** [comment]
Hi,

This notebook should work with the latest version of Sionna.
Unfortunately, I cannot reproduce the error.

Could you provide an example code snippet that reproduces the error?

**Reply by PudongDacihua** [reply]
Hi,
I've identified the problem. It seems that during channel equalization, the matrix_inv function in tensor.py is unable to successfully import config.py. This results in an error in the matrix_inv function due to the conditional statement:
if tensor.dtype in [tf.complex64, tf.complex128] \
    and sn.config.xla_compat \
    and not tf.executing_eagerly():
The temporary solution is to move all the channel estimation-related code into the main file, but it's not a long-term solution.

I believe this issue might be related to my simulation environment rather than the code itself. I will continue to work on resolving this problem.
